### PR TITLE
Allow connecting to robot with already authenticated BeamAPI instance.

### PR DESCRIPTION
**Pull request by jamesbirtles**
This allows for the use of oauth via the BeamAPI instance.

### CODE REVIEW

File: src/main/java/pro/beam/interactive/robot/RobotBuilder.java [hunk 1]
```diff
@@ -88,18 +88,22 @@ public RobotBuilder channel(BeamChannel channel) {
...
+     * If auth is set to true, then this method does, at some point, attempt to authenticate with the BeamAPI
+     * using the Beam API client, which will, in turn, tarnish its credentials. It is RECOMMENDED that a new instance
+     * (`new BeamAPI()`) be constructed and sent into this method's parameter list.
      *
      * @param beam An instance of the BeamAPI.
+     * @param auth Should the robot authenticate with the BeamAPI.
      * @return A ListenableFuture of type Robot that will return the connected, authenticated instance of the Robot.
      */
-    public ListenableFuture<Robot> build(final BeamAPI beam) {
+    public ListenableFuture<Robot> build(final BeamAPI beam, final boolean auth) {
```
Comment (ttaylorr) [inline src/main/java/pro/beam/interactive/robot/RobotBuilder.java]: How about `forceAuth`? Might be a bit clearer.

**Comment by ttaylorr** [comment]
Works for me, pending my one comment 👍

**Comment by TATDK** [comment]
Shouldn't this have been 1.6.0-SNAPSHOT instead of 1.5.1-SNAPSHOT as it adds new functionality?

As I assume the versioning of this API client follows the [Semantic Versioning](http://semver.org/)

> Given a version number MAJOR.MINOR.PATCH, increment the:
> MAJOR version when you make incompatible API changes,
> MINOR version when you add functionality in a backwards-compatible manner, and
> PATCH version when you make backwards-compatible bug fixes.

**Comment by ttaylorr** [comment]
Totally slipped by me! This should have been in 1.6.0-SNAPSHOT, but I'm going to leave it as is since I don't want to
1. unpublish 1.5.1-SNAPSHOT and break all clients of it
2. publish a duplicate version in 1.5.1 and 1.6.0-SNAPSHOT.

Thanks for letting me know!

**Pull request merged**